src/OpenTelemetry.Extensions.Hosting/OpenTelemetryServicesExtensions.cs

```diff
@@ -35,110 +37,37 @@ public static class OpenTelemetryServicesExtensions
         /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
         public static IServiceCollection AddOpenTelemetryTracing(this IServiceCollection services)
         {
-            services.AddOpenTelemetryTracing(builder => { });
-            return services;
-        }
-
-        /// <summary>
-        /// Adds OpenTelemetry TracerProvider to the specified <see cref="IServiceCollection" />.
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
-        /// <param name="configure">The <see cref="TracerProviderBuilder"/> action to configure TracerProviderBuilder.</param>
-        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
-        public static IServiceCollection AddOpenTelemetryTracing(this IServiceCollection services, Action<TracerProviderBuilder> configure)
-        {
-            if (configure is null)
+            services.AddSingleton<TracerProvider>((s) =>
             {
-                throw new ArgumentNullException(nameof(configure));
-            }
-
-            var builder = Sdk.CreateTracerProviderBuilder();
-            configure(builder);
-            services.AddOpenTelemetryTracing(() => builder.Build());
-            return services;
-        }
+                var builder = Sdk.CreateTracerProviderBuilder();
+                var configs = s.GetRequiredService<IEnumerable<IConfigureOptions<TracerProviderBuilder>>>();
+                foreach (var config in configs)
+                {
+                    config.Configure(builder);
+                }
 
-        /// <summary>
-        /// Adds OpenTelemetry TracerProvider to the specified <see cref="IServiceCollection" />.
-        /// </summary>
-        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
-        /// <param name="configure">The <see cref="TracerProviderBuilder"/> action to configure TracerProviderBuilder.</param>
-        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
-        public static IServiceCollection AddOpenTelemetryTracing(this IServiceCollection services, Action<IServiceProvider, TracerProviderBuilder> configure)
```

Comment:
New Relic's exporter uses `ILogger` for diagnostic logging. We use this overload that accepts an `IServiceProvider` in our examples to demonstrate how to configure our exporter with logging enabled. Is there a better way to achieve this?

See: https://github.com/newrelic/newrelic-telemetry-sdk-dotnet/blob/281223149ad73ac94f87fd7eafc41fa278bdbb1d/examples/NewRelic.OpenTelemetry/AspNetCore/Startup.cs#L25-L40